x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/alerts.ts

```diff
@@ -548,35 +506,18 @@ export default function alertTests({ getService }: FtrProviderContext) {
         });
 
         it('should reset throttle window when not firing', async () => {
-          const reference = `create-test-7:${user.username}`;
-          const createdAction = await createIndexRecordAction(space.id);
-          const response = await supertestWithoutAuth
-            .post(`${getUrlPrefix(space.id)}/api/alert`)
-            .set('kbn-xsrf', 'foo')
-            .auth(user.username, user.password)
-            .send(
-              getTestAlertData({
-                interval: '1s',
-                throttle: '1m',
-                alertTypeId: 'test.always-firing',
-                alertTypeParams: {
-                  index: ES_TEST_INDEX_NAME,
-                  reference,
-                  groupsToScheduleActionsInSeries: ['default', null, 'default'],
-                },
-                actions: [
-                  {
-                    group: 'default',
-                    id: createdAction.id,
-                    params: {
-                      index: ES_TEST_INDEX_NAME,
-                      reference,
-                      message: 'from:default',
-                    },
-                  },
-                ],
-              })
-            );
+          const reference = alertUtils.generateReference();
+          const response = await alertUtils.createAlwaysFiringAction({
+            reference,
+            overwrites: {
+              interval: '1s',
+              alertTypeParams: {
+                index: ES_TEST_INDEX_NAME,
+                reference,
+                groupsToScheduleActionsInSeries: ['default', null, 'default'],
+              },
+            },
+          });
 
           switch (scenario.id) {
```

Comment:
I realise this isn't a new pattern, as there was already a preexisting switch like this above, but this pattern feels a little dangerous to me.

It feels like we're packing multiple tests into one and I can't help but wonder if we're stretching the scenario pattern here too much.
As a person new to this test suite, following this is hard and confusing and defeats one of the purposes of testing which is providing a clear documentation of how an API is meant to be used and what the expected behaviour is.

I wonder if perhaps we can have the tests with the Switch case explicit for each scenario, and only use the `for` loop for tests that are the same for each scenario and don't require a switch?

This isn't a blocker for me, but I can't shale the feeling this should be rethought.

Reply:
To me if I call an API to put something a state, and it's already in that state, that's not an error, so it makes sense to just say "OK" in that case.  What value would returning an error response give?  The client would then have to figure out what state things are in, and oh yeah, they're in the state the client wanted anyway.  Seems like a distinction without a difference to me.

Reply:
To provide a bit of background on the tests. These were inspired from https://github.com/elastic/kibana/tree/master/x-pack/test/ui_capabilities as suggested by the security team. We need some coverage when security is enabled / disabled as well as spaces enabled / disabled. In order to just have coverage we opted to follow the approach they did and defer until later a look at how we want our test suite to work (https://github.com/elastic/kibana/projects/26#card-26655072). 

There will be room to propose different approaches but yeah I would say lets talk then.

Reply:
@bmcconaghy From my perspective that would be an issue in how the error response is communicated, not with whether an error response is valid. 🤷‍♂️

I dislike No-ops, it's subjective, but they often cause invalid states to be missed so I try to avoid them when possible.
But like I said, by no means a blocker.